paddle/fluid/framework/ir/graph.h

```diff
@@ -252,11 +258,29 @@ class Graph {
   std::shared_ptr<Graph> Clone();
 
  private:
+#if FLAGS_ssa_program
+  std::map<std::string, std::vector<ir::Node *>> InitFromBlock(
+      const BlockDesc &block);
+
+  void ReleaseSubgraphs() { sub_graphs_.clear(); }
+
+  void AddSubgraph(std::unique_ptr<Graph> sub_graph) {
+    sub_graphs_.push_back(std::move(sub_graph));
+  }
+
+  std::unique_ptr<Graph> CloneSubgraph(const size_t idx);
+#else   // FLAGS_ssa_program
   std::map<std::string, std::vector<ir::Node *>> InitFromProgram(
       const ProgramDesc &program);
+#endif  // FLAGS_ssa_program
 
   // NOTE: program_ shouldn't be exposed to user.
   const ProgramDesc program_;
+#if FLAGS_ssa_program
+  const Graph *parent_;  // not owned.
```

Comment:
Call it parent may not be so good.

In program, global block is special. Can we call it `global_graph_`?

Reply:
As far as I know, there are at least two global programs in the ladder runtime. I think it would lead to some misunderstanding.

Another consideration is that, as the basic function of converting program into graph, I think it could be better to regard it as an independent function rather than limited to the case of global program to global graph.